card/mod/01_core/set/all/name.rb

```diff
@@ -262,26 +280,29 @@ def suspend_name(name)
     Rails.logger.info "cascading name: #{dep.name}"
     Card.expire dep.name #old name
     newname = dep.cardname.replace_part name_was, name
-    Card.where( id: dep.id ).update_all name: newname.to_s, key: newname.key
+    Card.where(id: dep.id).update_all name: newname.to_s, key: newname.key
     Card::Reference.update_on_rename dep, newname, update_referencers
     Card.expire newname
   end
-  if update_referencers
-    Auth.as_bot do
-      [self.name_referencers(name_was)+(deps.map &:referencers)].flatten.uniq.each do |card|
-        # FIXME  using "name_referencers" instead of plain "referencers" for self because there are cases where trunk and tag
-        # have already been saved via association by this point and therefore referencers misses things
-        # eg.  X includes Y, and Y is renamed to X+Z.  When X+Z is saved, X is first updated as a trunk before X+Z gets to this point.
-        # so at this time X is still including Y, which does not exist.  therefore #referencers doesn't find it, but name_referencers(old_name) does.
-        # some even more complicated scenario probably breaks on the dependents, so this probably needs a more thoughtful refactor
-        # aligning the dependent saving with the name cascading
-
-        Rails.logger.debug "------------------ UPDATE REFERER #{card.name}  ------------------------"
-        unless card == self or card.structure
-          card = card.refresh
-          card.db_content = card.replace_references name_was, name
-          card.save!
-        end
+  execute_referencers_update(deps) if update_referencers
+end
+
+def execute_referencers_update dependents
+  Auth.as_bot do
+    [self.name_referencers(name_was) + dependents.map(&:referencers) ]
+      .flatten.uniq.each do |card|
+      # FIXME  using 'name_referencers' instead of plain 'referencers' for self because there are cases where trunk and tag
```

Comment:
Line is too long. [123/80]<br>Annotation keywords like `FIXME` should be all upper case, followed by a colon, and a space, then a note describing the problem.